source/extensions/bootstrap/internal_listener/internal_listener_registry.cc

```diff
@@ -13,9 +13,16 @@ namespace InternalListener {
 SINGLETON_MANAGER_REGISTRATION(internal_listener_registry);
 
 InternalListenerExtension::InternalListenerExtension(
-    Server::Configuration::ServerFactoryContext& server_context)
+    Server::Configuration::ServerFactoryContext& server_context,
+    const envoy::extensions::bootstrap::internal_listener::v3::InternalListener& config)
     : server_context_(server_context),
       tls_registry_(std::make_shared<TlsInternalListenerRegistry>()) {
+  // Convert buffer size proto config to KB. If configuration is zero or not specified,
+  // assign default buffer size to be 1MB.
+  buffer_size_ = config.buffer_size().value() * 1024;
+  if (buffer_size_ == 0) {
+    buffer_size_ = INTERNAL_CONNECTION_DEFAULT_BUFFER_SIZE;
+  }
```

Comment:
nit: use `has_buffer_size()`.
If `0` is an invalid value, the config should be rejected.

Reply:
Yeah, added the lower bound in proto config.